tests/test_clear_stream_source.py

```
@@ -206,3 +206,31 @@ def test_arguments_are_validated_and_close_stops_everything(self):
         source, popen = self.source(lambda command: FakeProcess())
         source.close()
         self.assertFalse(source.start_session(), "a closed source starts no session")
+
+    def test_a_fresh_session_with_no_live_frame_yet_still_decodes_the_buffered_keyframe(self):
+        decoded = jpeg()
+        never = Event()
+
+        class SilentStdout:
+            def read(self, _size):
+                never.wait(0.05)
+                return b"" if never.is_set() else b""
```

Ruff (0.16.3)

[warning] Useless `if`-`else` condition

(RUF034)